iroha/src/wsv.rs

```diff
@@ -89,3 +99,46 @@ impl WorldStateView {
         self.domain(&id.domain_name)?.asset_definitions.get_mut(id)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::{
+        block::BlockHeader,
+        peer::{Peer, PeerId},
+    };
+    use std::collections::{HashMap, HashSet};
+
+    #[async_std::test]
+    async fn test_listeners() {
+        let block = CommittedBlock {
+            header: BlockHeader {
+                timestamp: 0,
+                height: 0,
+                previous_block_hash: [0; 32],
+                merkle_root_hash: [0; 32],
+            },
+            transactions: Vec::new(),
+            signatures: Vec::new(),
+        };
+        let listeners = vec![Instruction::If(
+            Box::new(Instruction::Notify("Test".to_string())),
+            Box::new(Instruction::Notify("Passed".to_string())),
+            None,
+        )];
+        let mut world_state_view = WorldStateView {
+            peer: Peer {
+                id: PeerId {
+                    address: "127.0.0.1".to_string(),
+                    public_key: [0; 32],
+                },
+                peers: HashSet::new(),
+                listen_address: "127.0.0.1".to_string(),
+                domains: HashMap::new(),
+                listeners,
+            },
+            blocks: Vec::new(),
+        };
+        world_state_view.put(&block).await;
```

Comment:
We can see in the output that the test failed or passed, but it will not fail the pipeline. And if we take into account the amount of prints our tests usually produce, the result of this one will be practically invisible.

I think we should find a way to check if the listener instruction was really executed and if it was a correct one. Maybe use an ISI that changes the state of WSV here.